### PR TITLE
Adding support for pretty-printing JSON results

### PR DESCRIPTION
**Pull request by moozzyk**
Fixes: #6246

This change adds a new output mode that outputs JSON results in a more
readable format (a.k.a. pretty print). The existing json output mode
remains untouched.

Test Plan
```
./osquery/osqueryi --help
...
    --json                                           Set output mode to 'json'
    --json_pretty                                    Set output mode to 'json_pretty'
...
```

```
./osquery/osqueryi --json_pretty "SELECT * FROM routes WHERE destination = '::1'"
[
  {
    "destination": "::1",
    "flags": "2098181",
    "gateway": "::1",
    "hopcount": "0",
    "interface": "lo0",
    "metric": "0",
    "mtu": "16384",
    "netmask": "128",
    "source": "",
    "type": "local"
  }
]
```

```
./osquery/osqueryi --json "SELECT * FROM routes WHERE destination = '::1'"
[
  {"destination":"::1","flags":"2098181","gateway":"::1","hopcount":"0","interface":"lo0","metric":"0","mtu":"16384","netmask":"128","source":"","type":"local"}
]
```

- added tests
- formatting tests pass

### CODE REVIEW

**Comment by theopolis** [comment]
Hi @moozzyk, thanks for the PR. I think this looks good but you'll need to sign the ICLA or CCLA, by clicking the link posted by the CLA bot, before we can merge.

**Comment by moozzyk** [comment]
Thanks @theopolis! I am working on that - I did not notice that this repo was moved out of facebook org and has now a different CLA so I need to get an approval. I will sign the CLA as soon as I can.

**Comment by moozzyk** [comment]
I just signed the CLA.

**Review by theopolis — APPROVED** [review]

**Pull request merged**
